### PR TITLE
feat(nx): init for use nx

### PR DESCRIPTION
**Pull request by LinboLen**
Issue Number: N/A

## PR Details

related https://github.com/amplication/amplication/issues/3272

## PR Checklist
- [ ] Tests for the changes have been added
- [ ] `npm test` doesn't throw any error

IMPORTANT: Please review the [CONTRIBUTING.md](https://github.com/amplication/amplication/blob/master/CODE_OF_CONDUCT.md) file for detailed contributing guidelines.

### CODE REVIEW

**Comment by LinboLen** [comment]
create pull request and closed first. in order to take a pull request mark

**Comment by yuval-hazaz** [comment]
Thanks @LinboLen.
 @arielweinberger is leading the efforts to migrate to nx, so I added home as a reviewer. He probably will see it on Monday since he is away for a few days.

**Comment by LinboLen** [comment]
feel glad that amplication can use nx to build. there are still some bugs need to solve

**Comment by arielweinberger** [comment]
Hi @LinboLen - first of all I want to thank you for taking on this change. Moving to Nx is definitely something I want to do.

It will take me a day or two before I can properly review this PR and test your changes.

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,4 @@
+
+[*]
+indent_style = space
+tab_width = 2
```
Comment (arielweinberger) [inline .editorconfig:4]: This change is not related to the Nx migration I believe. Let's remove it.

File: .env [hunk 2]
```diff
@@ -1,8 +1,51 @@
...
-SERVICE_JWT_SECRET=service-jwt-secret
+SERVICE_DB_NAME=amplication-server
+
+# Nest
+CORS_ENABLE=1
+PORT=3000
+HOST=http://localhost:3000
+CLIENT_HOST=http://localhost:3001
+
+# GraphQL
```
Comment (arielweinberger) [inline .env:18]: I see a large amount of changes to the root-level `.env` which isn't per se related to the Nx migration.

I would like to keep it at envfiles per app.

**Comment by LinboLen** [comment]
nest server can run properly. I don't know why

File: packages/amplication-client/src/CommandPalette/CommandPalette.tsx [hunk 3]
```diff
@@ -122,8 +122,9 @@ const CommandPalette = ({ trigger }: Props) => {
   const { data } = useQuery<TData>(SEARCH, {
     variables: { query, projectId: currentProject?.id },
   });
-  const commands = useMemo(
+    const commands = useMemo(
     () =>
+  // @ts-ignore
```
Comment (arielweinberger) [inline packages/amplication-client/src/CommandPalette/CommandPalette.tsx:127]: I prefer to fix the root cause, and generaly we don't accept or push any additional `// @ts-ignore` statements, we see it as a bad practice.

File: packages/amplication-client/src/Entity/DataTypeSelectField.tsx [hunk 4]
```diff
@@ -42,6 +42,7 @@ const DataTypeSelectField = (props: Props) => {
       const defaultValues = Object.fromEntries(
         Object.entries(schema.properties).map(([name, property]) => [
           name,
+            // @ts-ignore
```
Comment (arielweinberger) [inline packages/amplication-client/src/Entity/DataTypeSelectField.tsx:45]: Please remove `// @ts-ignore` and fix the root cause.

File: packages/amplication-client/.flowconfig [hunk 5]
```diff
@@ -0,0 +1,3 @@
+[options]
+enums=true
+type=true
```
Comment (arielweinberger) [inline packages/amplication-client/.flowconfig:3]: This change is not related to the Nx migration, please remove it.

File: packages/amplication-client/src/Components/EditableTitleField.tsx [hunk 6]
```diff
@@ -1,9 +1,9 @@
 import React from "react";
-import { TextField, Props } from "@amplication/design-system";
+import { TextField, TextFieldProps } from "@amplication/design-system";
```
Comment (arielweinberger) [inline packages/amplication-client/src/Components/EditableTitleField.tsx:2]: Why did you rename `Props ` to `TextFieldProps`? I don't believe this is essential for the Nx migration.

Our repo has **a lot of traffic** and changes like this can easily cause a bunch of conflicts.

File: packages/amplication-client/src/Entity/SchemaField.tsx [hunk 7]
```diff
@@ -66,6 +66,7 @@ export const SchemaField = ({
         throw new Error("Array schema must define items");
       }
 
+      // @ts-ignore
```
Comment (arielweinberger) [inline packages/amplication-client/src/Entity/SchemaField.tsx:69]: Please remove `// @ts-ignore` and fix the root cause.

File: packages/amplication-client/src/Layout/MenuItem.tsx [hunk 8]
```diff
@@ -49,6 +49,7 @@ const MenuItem = ({
       buttonStyle={EnumButtonStyle.Text}
       as={to ? NavLink : Button}
       onClick={onClick}
+        // @ts-ignore
```
Comment (arielweinberger) [inline packages/amplication-client/src/Layout/MenuItem.tsx:52]: Please remove `// @ts-ignore` and fix the root cause.

File: packages/amplication-server/migration-scripts/delete-auto-number.ts [hunk 9]
```diff
@@ -4,8 +4,9 @@ import { PrismaClient, EnumDataType } from '@amplication/prisma-db';
 
 async function main() {
   const client = new PrismaClient();
-  await client.entityField.deleteMany({
+	await client.entityField.deleteMany({
     where: {
+  // @ts-ignore
```
Comment (arielweinberger) [inline packages/amplication-server/migration-scripts/delete-auto-number.ts:9]: Please remove `// @ts-ignore` and fix the root cause.

File: packages/amplication-server/migration-scripts/create-project-configuration.ts [hunk 10]
```diff
@@ -70,6 +70,7 @@ async function main() {
               displayName: DEFAULT_PROJECT_CONFIGURATION_SETTINGS_NAME,
               resource: {
                 create: {
+					// @ts-ignore
```
Comment (arielweinberger) [inline packages/amplication-server/migration-scripts/create-project-configuration.ts:73]: Please remove `// @ts-ignore` and fix the root cause.

**Review by arielweinberger — CHANGES_REQUESTED** [review]

**Comment by arielweinberger** [comment]
@LinboLen I took some time to review things at a high level in the meantime. I have two main points:
- I see a lot of usage of `@ts-ignore`, which is not something we can accept. We would need to fix the root cause.
- I see a lot of changes to import paths (from `src` to relative), which is 100% correct and I am very happy you did that. May I suggest putting the import path changes in a separate PR? Then we can review it quickly, merge it, and keep the Nx PR to focus on the Nx changes rather than file imports. This will reduce the range of impact.

**Comment by LinboLen** [comment]
@arielweinberger thank you take time to review it. there still are some of problem need to be sovled.  this pr is stiil should be closed state. 
my notebook lack of disk usage so I want to use nx to build. 

closed now.
